### ISSUE TITLE
az costmanagement query does not provide a way to overwrite property Granularity

### ISSUE THREAD

**Issue by niswitze** [Azure/azure-cli-extensions#1935]
- If the issue is to do with Azure CLI 2.0 in-particular, create an issue here at [Azure/azure-cli](https://github.com/Azure/azure-cli/issues)

### Extension name (the extension in question)
costmanagement

### Description of issue (in as much detail as possible)
az costmanagement query does not provide a way to overwrite property Granularity. It is set to 'Daily' by default and the extension should allow the option to change this. 

The request body for all query based calls to the cost management APIs resemble like the following:

{"type": "Usage", "timeframe": "MonthToDate", "dataSet": {"granularity": "Daily", "configuration": {}}}
-----

**Comment by yonzhan** [comment]
add to S173

**Comment by haroldrandom** [comment]
It's a constant defined in QueryDefinition. If need change, have to inform awear of service team to get support.

**Comment by ramaganesan-rg** [comment]
Thank you for your feedback. Team is investigating this.

**Comment by jochav** [comment]
Hi @niswitze, sorry for the delay, we are in the process of validating this feature request. Will come back to you soon with a decision and possible ETA.

**Comment by malikbilal27** [comment]
Any update here ?

**Comment by niswitze** [comment]
> Any update here ?

@jochav bumping the thread here as well.

**Comment by jochav** [comment]
Yes, sorry for the delay, needed to verify with a few people. But apparently there was a gap in my understanding of the problem. As far as we can tell, Query API does allow to change the granularity, but the documentation and likely SDK is not displaying this. Will work next to update these.
